src/main/java/net/dv8tion/jda/internal/requests/restaction/WebhookMessageActionImpl.java

```diff
@@ -0,0 +1,253 @@
+    {
+        Checks.notNull(message, "Message");
+        this.tts = message.isTTS();
+        this.embeds.addAll(message.getEmbeds());
+        this.allowedMentions.applyMessage(message);
+        this.components.addAll(message.getActionRows());
+        return setContent(message.getContentRaw());
+    }
+
+    @Nonnull
+    @Override
+    public WebhookMessageActionImpl<T> setEphemeral(boolean ephemeral)
+    {
+        this.ephemeral = ephemeral;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public WebhookMessageActionImpl<T> setContent(@Nullable String content)
+    {
+        this.content.setLength(0);
+        if (content != null)
+            this.content.append(content);
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public WebhookMessageActionImpl<T> setTTS(boolean tts)
+    {
+        this.tts = tts;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public WebhookMessageActionImpl<T> setUsername(@Nullable String name)
+    {
+        if (name != null)
+        {
+            Checks.notEmpty(name, "Name");
+            Checks.notLonger(name, 128, "Name");
+        }
+        this.username = name;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public WebhookMessageActionImpl<T> setAvatarUrl(@Nullable String iconUrl)
+    {
+        if (iconUrl != null && iconUrl.isEmpty())
+            iconUrl = null;
+        this.avatarUrl = iconUrl;
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public WebhookMessageActionImpl<T> addEmbeds(@Nonnull Collection<? extends MessageEmbed> embeds)
+    {
+        Checks.noneNull(embeds, "Message Embeds");
+        Checks.check(this.embeds.size() + embeds.size() <= 10, "Cannot have more than 10 embeds in a message!");
+        this.embeds.addAll(embeds);
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public WebhookMessageActionImpl<T> addFile(@Nonnull InputStream data, @Nonnull String name, @Nonnull AttachmentOption... options)
+    {
+        Checks.notNull(name, "Name");
+        Checks.notNull(data, "Data");
+        Checks.notNull(options, "AttachmentOption");
+        // Yes < 10 not <= 10 since we add one after this
+        Checks.check(files.size() < 10, "Cannot have more than 10 files in a message!");
+        if (options.length > 0 && options[0] == AttachmentOption.SPOILER)
+            name = "SPOILER_" + name;
+        files.put(name, data);
+        return this;
+    }
+
+    @Nonnull
+    @Override
+    public WebhookMessageActionImpl<T> addActionRows(@Nonnull ActionRow... rows)
+    {
+        Checks.noneNull(rows, "ActionRows");
+        Checks.check(rows.length + components.size() <= 5, "Can only have 5 action rows per message!");
+        Collections.addAll(components, rows);
+        return this;
+    }
+
+    private DataObject getJSON()
+    {
+        DataObject json = DataObject.empty();
+        json.put("content", content.toString());
```

Comment:
Is it possible that this is why that `editOriginalComponents` doesn't work? We're always sending a string, even when it is empty (`""`)

For `MessageActionImpl` when `content.length() == 0` we `putNull("content")`

Reply:
That is only because MessageActionImpl deals with edits. This class is not for edits.